### DISCUSSION TITLE
Serverless Functions (Node.js and Python)

### DISCUSSION THREAD

**Discussion by nasserakhter** [coollabsio/coolify#2882] [Feature Requests]
Hi,

I would like to suggest adding support for serverless Node.js or Python functions in Coolify.

The idea is to enable connecting a Git project containing multiple individual JS files with functions. With a new built-in serverless feature, Coolify could use a single Nginx instance to manage and invoke various unrelated serverless functions.

Currently, I need to use a custom JS script to create an Express server for each of my repositories. Integrating this functionality into Coolify would streamline the process and reduce the overhead.

It would essentially mimic the functionality of Vercel / Next.js's serverless functions

Thank you!

**Comment by frederikheld** [comment]
I'd need serverless functions on Coolify myself right now. But after thinking about it, I don't think that Coolify would be the right place to put it. Coolify itself manages the applications you want to deploy. "something that runs serverless functions" is one application that can be deployed. I'm still looking for the right "something" to run functions but I don't think that serverless functions should be feature of Coolify.

**Comment by dimabran** [comment]
So you want to run your serverless applications on your server?

**Reply by nasserakhter** [reply]
Serverless in the sense that you have a js script that is called whenever a rest endpoint is invoked. Instead of a js script that is running a express application 24/7.

But like @frederikheld said, it makes sense for it instead to be a app that runs on top of coolify.

**Reply by frederikheld** [reply]
@dimabran That's what serverless functions are. It's just someone else's server. How do you think those functions work? Some process has to be running and waiting for requests. There's always a server involved. Serverless just means that you don't need to think about that server, which makes some things simpler (but others more complicated). The main difference between server and serverless is the lifetime of processes and the associated handling of state. It's a conceptual difference.

**Comment by adiologydev** [comment]
I am not sure about Python just yet but I may have a solution for this for JavaScript once one of my PRs gets merged. It won't be a major change to the platform, keeping the surface vector as minimal as possible while providing maximum value.

I'll keep ya'll posted!

**Reply by RayBB** [reply]
@adiologydev what PR is that?
